app/src/main/java/net/gsantner/markor/activity/SettingsActivity.java

```diff
@@ -199,10 +199,10 @@ public void doUpdatePreferences() {
             }
 
             setPreferenceVisible(R.string.pref_key__is_multi_window_enabled, Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP);
-            setPreferenceVisible(R.string.pref_key__default_encryption_password, Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT);
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && _as.hasPasswordBeenSetOnce()) {
-                updateSummary(R.string.pref_key__default_encryption_password, "****");
-                setDialogMessage(R.string.pref_key__default_encryption_password, getString(R.string.password_already_set_setting_a_new_password_will_overwrite));
+
+            setPreferenceVisible(R.string.pref_key__encryption_password, Build.VERSION.SDK_INT >= Build.VERSION_CODES.M);
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && !TextUtils.isEmpty(_as.getPassword())) {
+                updateSummary(R.string.pref_key__encryption_password, getString(R.string.hidden_password));
```

Comment:
@harshad1  / @opensource21  
Really required to change the settings key here? Want to make sure people not get their password reset somehow.  
Also I'd say `default_..._password` is still a valid name to use, because that is what it is - default password for everything.  
There is user interest for entering the encryption password per-file (`Per File Settings`) and/or forgetting the general password. 

Thus, we might have in future `Per-File password` and `Global default password`, similar to `Syntax Highlighting, Format Per-File`.

Reply:
I have reverted the key used to store the password to `pref_key__default_encryption_password`.